### PR TITLE
Add support for `breached_password_detection.stage` on `auth0_attack_protection` resource

### PR DESCRIPTION
**Pull request by nialdaly**
### Summary
This PR aims to close this [issue](https://github.com/auth0/terraform-provider-auth0/issues/386).

### CODE REVIEW

**Comment by sergiught** [comment]
@nialdaly You're a rockstar ⭐ Much appreciated for already trying to contribute this to the repo. Please don't hesitate to ask for support if needed to get this in the right place. 

Adding tests and making them pass for terraform will be very similar to how they are done on the Go SDK side: https://github.com/auth0/go-auth0/blob/main/CONTRIBUTING.md (I have on my TODO to update it here as well). 

Also make sure to update the examples and to regenerate the docs afterwards with `make docs`.

**Comment by nialdaly** [comment]
Hey @sergiught thanks for the pointers! I've updated the test recordings, not 100% sure if they are correct.

**Comment by nialdaly** [comment]
When running the tests locally @sergiught I'm getting the following error. I'm not sure if its related to the updated test recording:
```
=== RUN   TestAccAttackProtectionBreachedPasswordDetection
    resource_auth0_attack_protection_test.go:60: Step 2/4 error: After applying this test step and performing a `terraform refresh`, the plan was not empty.
        stdout
        
        
        Terraform used the selected providers to generate the following execution
        plan. Resource actions are indicated with the following symbols:
          ~ update in-place
        
        Terraform will perform the following actions:
        
          # auth0_attack_protection.my_protection will be updated in-place
          ~ resource "auth0_attack_protection" "my_protection" {
                id = "terraform-20230127163011958200000006"
        
              ~ breached_password_detection {
                  ~ shields                      = [
                      + "admin_notification",
                      + "block",
                    ]
                    # (3 unchanged attributes hidden)
        
                    # (1 unchanged block hidden)
                }
        
        
                # (2 unchanged blocks hidden)
            }
        
        Plan: 0 to add, 1 to change, 0 to destroy.
--- FAIL: TestAccAttackProtectionBreachedPasswordDetection (1.20s)
```

File: test/data/recordings/TestAccAttackProtectionBreachedPasswordDetection.yaml [hunk 1]
```diff
@@ -1254,7 +1254,7 @@ interactions:
       trailer: { }
       content_length: -1
       uncompressed: true
-      body: '{"enabled":false,"shields":["admin_notification","user_notification","block"],"admin_notification_frequency":["weekly","monthly","daily","immediately"],"method":"standard","stage":{"pre-user-registration":{"shields":[]}}}'
+      body: '{"enabled":false,"shields":["admin_notification","user_notification","block"],"admin_notification_frequency":["weekly","monthly","daily","immediately"],"method":"standard","stage":{"pre-user-registration":{"shields":["admin_notification"]}}}'
```
Comment (sergiught) [inline test/data/recordings/TestAccAttackProtectionBreachedPasswordDetection.yaml:1257]: This is getting updated on the wrong interaction @nialdaly, if you want I can push the regenerated recording.

Comment (nialdaly) [reply test/data/recordings/TestAccAttackProtectionBreachedPasswordDetection.yaml:1257]: That would be great!

Comment (sergiught) [reply test/data/recordings/TestAccAttackProtectionBreachedPasswordDetection.yaml:1257]: Alright, one moment ⌛

File: docs/resources/attack_protection.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ Optional:
 
 Optional:
 
-- `shields` (Set of String) Action to take when a breached password is detected during a signup.
+- `shields` (Set of String) Action to take when a breached password is detected during a signup. Possible values: `block`, `admin_notification`.
```
Comment (sergiught) [inline docs/resources/attack_protection.md:81]: Just making sure we also give the options within the docs.

File: internal/provider/resource_auth0_attack_protection.go [hunk 3]
```diff
@@ -396,6 +396,7 @@ func flattenBreachedPasswordProtection(bpd *management.BreachedPasswordDetection
 			"enabled":                      bpd.GetEnabled(),
 			"method":                       bpd.GetMethod(),
 			"admin_notification_frequency": bpd.GetAdminNotificationFrequency(),
+			"shields":                      bpd.GetShields(),
```
Comment (sergiught) [inline internal/provider/resource_auth0_attack_protection.go:399]: @nialdaly This is what was causing the tests to fail 🙈 we erroneously removed the shields property when we added the pre_user_registration block

File: internal/provider/resource_auth0_attack_protection_test.go [hunk 4]
```diff
@@ -40,7 +40,7 @@ resource "auth0_attack_protection" "my_protection" {
 		method = "standard"
 
 		pre_user_registration {
-			shields = ["admin_notification"]
+			shields = ["block", "admin_notification"]
```
Comment (sergiught) [inline internal/provider/resource_auth0_attack_protection_test.go:43]: Setting this with all the available options for the last test

File: test/data/recordings/TestAccAttackProtectionBreachedPasswordDetection.yaml [hunk 5]
```diff
@@ -19,7 +19,7 @@ interactions:
         Content-Type:
           - application/json
         User-Agent:
-          - Go-Auth0-SDK/0.12.0
+          - Go-Auth0-SDK/0.15.0
```
Comment (sergiught) [inline test/data/recordings/TestAccAttackProtectionBreachedPasswordDetection.yaml:22]: Updating the recording with AUTH0_HTTP_RECORDINGS=on. I'll make sure to update the contributing section as well (probably next week) for terraform like we did for go-auth0 to make it easier and have some handy make commands to regenerate automatically just for 1 test @nialdaly

**Review by sergiught — APPROVED** [review]
Great work @nialdaly 💪🏻 Fantastic contribution!

File: internal/provider/resource_auth0_attack_protection.go (hunk 3 above)
Comment (nialdaly) [reply internal/provider/resource_auth0_attack_protection.go:399]: ah my bad! Good catch haha

**Comment by nialdaly** [comment]
Thanks for all your help! @sergiught

**Pull request merged**